### DISCUSSION TITLE
[Task]:How to get predict results and modify them?

### DISCUSSION THREAD

**Discussion by lnbxldn** [open-edge-platform/anomalib#961] [Q&A]
### What is the motivation for this task?

How to reshape the inference result to origin shape and get the predict boundarys results and modify them?

### Describe the solution you'd like

I use reverse_distillation model on my own dataset, and i wanna inference the model and get the predict results, I use the command:
python tools/inference/lightning_inference.py 
	--config anomalib/models/reverse_distillation/config.yaml 
	--weights results/reverse_distillation/mvtec/my_dataset/run/weights/model.ckpt     
	--input datasets/MVTec/my_dataset/test/anomaly/   
	--output results/reverse_distillation/mvtec/my_dataset/run/images
the result image are transformed as the shape 800 x 800 and the color is green.
How to reshape the image result to origin shape and get the predict boundarys results and modify them?

**Comment by ashwinvaidya17** [comment]
@lnbxldn The `lightning_inference` relies on the [`ImageVisualizer` callback](https://github.com/openvinotoolkit/anomalib/blob/main/src/anomalib/utils/callbacks/visualizer/visualizer_image.py#L20). This relies on the [`Visualizer` class](https://github.com/openvinotoolkit/anomalib/blob/main/src/anomalib/post_processing/visualizer.py). You can modify the code within these files to achieve your requirements. If you want the anomaly maps to be of the same size as the input image. You can look into adding a reshape here https://github.com/openvinotoolkit/anomalib/blob/56706b3053cf27c9cd85b180511eb5b80256617e/src/anomalib/post_processing/visualizer.py#L105
Let us know if this solves your problem.

**Comment by LenaMetzen** [comment]
Hey, I modified the Lib/site-packages/pytorch_lightning/trainer.py file. In the _predict_impl function i extracted the values box_scores and box_labels and in a new function called _save_results i wrote both values in a .txt file.

```
def _predict_impl(
        self,
        model: Optional["pl.LightningModule"] = None,
        dataloaders: Optional[Union[EVAL_DATALOADERS, LightningDataModule]] = None,
        datamodule: Optional[LightningDataModule] = None,
        return_predictions: Optional[bool] = None,
        ckpt_path: Optional[str] = None,
    ) -> Optional[_PREDICT_OUTPUT]:
        Trainer._log_api_event("predict")
        log.detail(f"{self.__class__.__name__}: trainer predict stage")

        self.state.fn = TrainerFn.PREDICTING
        self.state.status = TrainerStatus.RUNNING
        self.predicting = True

        self.predict_loop.return_predictions = return_predictions  # type: ignore[assignment]

        # if a datamodule comes in as the second arg, then fix it for the user
        if isinstance(dataloaders, LightningDataModule):
            datamodule = dataloaders
            dataloaders = None
        if dataloaders is not None and datamodule:
            raise MisconfigurationException("You cannot pass both `trainer.predict(dataloaders=..., datamodule=...)`")

        if model is None:
            model = self.lightning_module
            model_provided = False
        else:
            model_provided = True

        # links data to the trainer
        self._data_connector.attach_data(model, predict_dataloaders=dataloaders, datamodule=datamodule)

        self._ckpt_path = self._checkpoint_connector._set_ckpt_path(
            self.state.fn, ckpt_path, model_provided=model_provided, model_connected=self.lightning_module is not None
        )

        self._predicted_ckpt_path = self.ckpt_path  # TODO: remove in v1.8

        results = self._run(model, ckpt_path=self.ckpt_path)

        assert self.state.stopped
        self.predicting = False

        for result in results:
            box_scores = result.get('box_scores', [])
            box_labels = result.get('box_labels', [])
            self._save_results(box_scores, box_labels)

        return results
    



    def _save_results(self, box_scores, box_labels) -> None:

        with open('results.txt', 'w') as file:
            file.write(str(box_scores))
            file.write("\n")
            file.write(str(box_labels))

        return
```
